src/OpenTelemetry.Exporter.Geneva/Metrics/OtlpProtobuf/OtlpProtobufMetricExporter.cs

```diff
@@ -15,17 +16,32 @@ internal sealed class OtlpProtobufMetricExporter : IDisposable
 
     private readonly Func<Resource> getResource;
 
-    public OtlpProtobufMetricExporter(Func<Resource> getResource, ConnectionStringBuilder connectionStringBuilder, IReadOnlyDictionary<string, object> prepopulatedMetricDimensions)
+    public OtlpProtobufMetricExporter(
+        Func<Resource> getResource,
+        ConnectionStringBuilder connectionStringBuilder,
+        IReadOnlyDictionary<string, object> prepopulatedMetricDimensions)
     {
+        Debug.Assert(getResource != null, "getResource was null");
+
+        this.getResource = getResource;
+
+#if NET6_0_OR_GREATER
+        IMetricDataTransport transport = !RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
+            ? MetricUnixUserEventsDataTransport.Instance
+            : MetricWindowsEventTracingDataTransport.Instance;
+#else
         if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
         {
-            // Temporary until we add support for user_events.
             throw new NotSupportedException("Exporting data in protobuf format is not supported on Linux.");
```

Comment:
don't we need to remove this?

Reply:
There is a block above this wrapped with `#if NET6_0_OR_GREATER` that's where user_events get lit up. Because LinuxTracepoints-Net only supports `net6.0`+: https://github.com/microsoft/LinuxTracepoints-Net/blob/974c47522d053c915009ef5112840026eaf22adb/Provider/Provider.csproj#L14

This block here (the original block) is left in for `net462` and `netstandard2.0` targets. The `net462` one should never be hit on linux. The `netstandard2.0` one _could_ be hit if someone forces a ref to the `netstandard2.0` dll. I wouldn't expect that to be common though. And if they do, this exception ensures they blow up right away.

Now we did vendor the code for LinuxTracepoints-Net so I _could_ attempt to make it work on `netstandard2.0` but I would prefer to make as few changes to the source files as possible to make it easy to upgrade to newer versions if needed.